src/error.rs

```diff
@@ -1176,6 +1176,33 @@ signed, unsigned, big, little, native, size, unit",
                     )
                     .unwrap();
                 }
+
+                NonExhaustiveLet {
+                    location,
+                    constructor,
+                    unhandled_constructors,
+                } => {
+                    let diagnostic = Diagnostic {
+                        title: "Non-exhaustive let binding".to_string(),
+                        label: "used here".to_string(),
+                        file: path.to_str().unwrap().to_string(),
+                        src: src.to_string(),
+                        location: location.clone(),
+                    };
+                    write(buffer, diagnostic, Severity::Error);
+                    writeln!(
+                        buffer,
+                        "You are handling the {} constructor, but this could also be one of:
+{}
+
+Either use a case expression to handle all possibilities, or use assert instead
+of let if you are sure you don't want to handle the others. This will cause a
```

Comment:
Could be a `try` rather than a `let`

Reply:
Might be worth having two different error messages then, one for let and one for try, as we can't really suggest using assert if you're currently using try.

Reply:
Oh yes, good point, I didn't think of that.

Reply:
Any preference between implementing two different errors altogether or just one with an extra binding type field and using that to show two different bits of text?

Reply:
Went with the latter